src/main/java/org/opensearch/sdk/sample/helloworld/ExampleCustomSettingConfig.java

```diff
@@ -17,16 +17,36 @@
  * {@link ExampleCustomSettingConfig} contains the custom settings value and their static declarations.
  */
 public class ExampleCustomSettingConfig {
-    private static final String FORBIDDEN_REGEX = "forbidden";
-
+    private static final String FORBIDDEN_REGEX = "^((?!forbidden).)*$";
```

Comment:
While I listed this as a possibility in the issue, the whole purpose of the boolean is so that we don't need these confusing regexes that nobody understands.

So I'd really like this kept as "forbidden" with the boolean set for it to NOT match (the false case).

For the URL check you can set the boolean to true.  This will effectively test both boolean values and you don't need to create a third one.

Reply:
can I change the validation condition in the validate() method to this :
 ` if (isMatching && value.contains(pattern.pattern())) {`, replacing matcher() with contains() check

Reply:
Can I use the contains() method in the validate() method in order not to compare through regular expressions, but to compare a word with a string? example :
`        public void validate(String value) {
            if (isMatching && value.contains(pattern.pattern())) {
                throw new IllegalArgumentException("Setting [" + value + "] does not match regex [" + pattern.pattern() + "]");
            } else if (!isMatching && !value.contains(pattern.pattern())) {
                throw new IllegalArgumentException("Setting [" + value + "] must match regex [" + pattern.pattern() + "]");
            }
        }`

Reply:
> Can I use the contains() method in the validate() method 

You could create a new `contains` validator, but it's no different than matching a regex.  That was the whole point of building this single validator.

Matching a string "foo" in a regex is pretty much exactly the same thing as `contains`.  If you want the entire word you can use beginning and end of line markers, etc.

We basically want two validators here and four tests:

1. A plain string "forbidden" (which happens to work out to the same as "must not contain" in a "must not match"), testing against one string containing the word (exception) and one not (does not throw).
2. A regex which "must match" testing against a string that matches (does not throw) and does not (exception)